src/main/java/io/confluent/connect/elasticsearch/ElasticsearchClient.java

```diff
@@ -362,23 +347,28 @@ public void afterBulk(long executionId, BulkRequest request, Throwable failure)
    * @return the return value of the called function
    */
   private <T> T callWithRetries(String description, Callable<T> function) {
-    try {
-      return RetryUtil.callWithRetries(
-          description,
-          function,
-          config.maxRetries(),
-          config.retryBackoffMs()
-      );
-    } catch (Exception e) {
-      throw new ConnectException("Failed to " + description + ".", e);
-    }
+    return RetryUtil.callWithRetries(
+        description,
+        function,
+        config.maxRetries() + 1,
```

Comment:
why +1 ?

Reply:
The argument, maxAttempts is the overall number of attempts, including the first call.
maxRetries can be 0 (no retries needed).

Reply:
got it, if maxAttempts isnt documented , could we please document it that it includes the first call

Reply:
Extended the current documentation and renamed the parameter for more clarity